src/Storages/MergeTree/registerStorageMergeTree.cpp

```diff
@@ -445,6 +446,14 @@ static StoragePtr create(const StorageFactory::Arguments & args)
     }
     else if (merging_params.mode == MergeTreeData::MergingParams::Replacing)
     {
+        if (arg_cnt && !engine_args[arg_cnt - 1]->as<ASTLiteral>())
+        {
+            if (!tryGetIdentifierNameInto(engine_args[arg_cnt - 1], merging_params.is_deleted_column))
+                throw Exception(
+                    "is_deleted column name must be an unquoted string" + getMergeTreeVerboseHelp(is_extended_storage_def),
+                    ErrorCodes::BAD_ARGUMENTS);
+            --arg_cnt;
+        }
```

Comment:
does mean that not backward compatibility of "ENGINE = ReplacingMergeTree(version)" ?
because the lasted arg is is_deleted_column now.